### PR TITLE
Making default Plugin configurations accessible when developing. 

### PR DESCRIPTION
**Pull request by BaronGreenback**
A configuration file is only ever created if the plugin provides a method of updating its settings.

This change ensures that when a plug configuration is created for the first time, its settings are stored to disk.

This should make it easier to develop plugins.

### CODE REVIEW

**Comment by crobibero** [comment]
I don't think this change is needed, if a plugin has a configuration that the user can edit the plugin should provide a way to change it without manually editing the file

**Comment by BaronGreenback** [comment]
I was thinking more from the dev side of things than the usrr. It means i can write the plugin settings class then concentrate and test plugin development and get that working before moving to add the web section.

Yes a could create a xml file manually but thats a pain and can be prone to errors and besides -  i'm lazy! Anything that can help speed up my dev and testing is good for me.

File: MediaBrowser.Common/Plugins/BasePlugin.cs [hunk 1]
```diff
@@ -252,20 +252,23 @@ private TConfigurationType LoadConfiguration()
...
+                return config;
             }
         }
 
         /// <summary>
         /// Saves the current configuration to the file system.
         /// </summary>
-        public virtual void SaveConfiguration()
+        /// <param name="config">Configuration to save.</param>
+        public virtual void SaveConfiguration(TConfigurationType config)
```
Comment (oddstr13) [inline MediaBrowser.Common/Plugins/BasePlugin.cs:265]: This function is called directly in some plugins.
Maybe add a function overload?
```
$ grep 'SaveConfiguration()' . -r
./jellyfin-plugin-intros/Jellyfin.Plugin.Intros/IntroManager.cs:            Plugin.Instance.SaveConfiguration();
./jellyfin-plugin-iptv/MediaBrowser.Channels.IPTV/Api/ServerApiEndpoints.cs:            Plugin.Instance.SaveConfiguration();
./jellyfin-plugin-trakt/Trakt/Api/ServerApiEndpoints.cs:                Plugin.Instance.SaveConfiguration();
./jellyfin-plugin-trakt/Trakt/Api/TraktApi.cs:                            Plugin.Instance.SaveConfiguration();
./jellyfin-plugin-trakt/Trakt/Api/TraktApi.cs:                Plugin.Instance.SaveConfiguration();
```

**Review by oddstr13 — CHANGES_REQUESTED** [review]

**Review by crobibero — APPROVED** [review]
Would be simpler to call the new function with the parameter

**Comment by oddstr13** [comment]
> Would be simpler to call the new function with the parameter

Yes, this is what I meant when I said overload – either way works, even if I think this makes for cleaner and more easily maintainable code

```csharp
public virtual void SaveConfiguration()
{
     SaveConfiguration(Configuration);
}
```

**Review by oddstr13 — APPROVED** [review]

**Comment by BaronGreenback** [comment]
> > Would be simpler to call the new function with the parameter
> 
> Yes, this is what I meant when I said overload – either way works, even if I think this makes for cleaner and more easily maintainable code
> 
> ```cs
> public virtual void SaveConfiguration()
> {
>      SaveConfiguration(Configuration);
> }
> ```

Brain fart - updated code.

**Pull request merged**